src/app/storage/FabricTableImpl.h

```diff
@@ -0,0 +1,140 @@
+    void Finish();
+
+    // Scene count
+    CHIP_ERROR GetEndpointEntryCount(uint8_t & entry_count);
+    CHIP_ERROR GetFabricEntryCount(FabricIndex fabric_index, uint8_t & entry_count);
+
+    // Data
+    CHIP_ERROR GetRemainingCapacity(FabricIndex fabric_index, uint8_t & capacity);
+    CHIP_ERROR SetTableEntry(FabricIndex fabric_index, const TableEntry & entry);
+    CHIP_ERROR GetTableEntry(FabricIndex fabric_index, StorageId entry_id, TableEntry & entry);
+    CHIP_ERROR RemoveTableEntry(FabricIndex fabric_index, StorageId entry_id);
+    CHIP_ERROR RemoveTableEntryAtPosition(EndpointId endpoint, FabricIndex fabric_index, EntryIndex entry_idx);
+
+    // Fabrics
+    CHIP_ERROR RemoveFabric(FabricIndex fabric_index);
+    CHIP_ERROR RemoveEndpoint();
+
+    void SetEndpoint(EndpointId endpoint);
+    void SetTableSize(uint16_t endpointEntryTableSize, uint16_t maxPerFabric);
+    bool IsInitialized() { return (mStorage != nullptr); }
+
+protected:
+    // gcc bug prevents us from using a static variable; see:
+    // https://stackoverflow.com/questions/50638053/constexpr-static-data-member-without-initializer
+    static constexpr EndpointId kEntryEndpointClusterID();
+
+    // This constructor is meant for test purposes, it allows to change the defined max for entries per fabric and global, which
+    // allows to simulate OTA where this value was changed
+    FabricTableImpl(uint16_t maxEntriesPerFabric, uint16_t maxEntriesPerEndpoint) :
+        mMaxPerFabric(maxEntriesPerFabric), mMaxPerEndpoint(maxEntriesPerEndpoint)
+    {}
+
+    // Global entry count
+    CHIP_ERROR SetEndpointEntryCount(const uint8_t & entry_count);
+
+    // wrapper function around emberAfGetClustersFromEndpoint to allow override when testing
+    virtual uint8_t GetClustersFromEndpoint(ClusterId * clusterList, uint8_t listLen);
+
+    // wrapper function around emberAfGetClusterCountForEndpoint to allow override when testing
+    virtual uint8_t GetClusterCountFromEndpoint();
+
+    class EntryIteratorImpl : public EntryIterator
+    {
+    public:
+        EntryIteratorImpl(FabricTableImpl & provider, FabricIndex fabricIdx, EndpointId endpoint, uint16_t maxEntriesPerFabric,
+                          uint16_t maxEntriesPerEndpoint);
+        size_t Count() override;
+        bool Next(TableEntry & output) override;
+        void Release() override;
+
+    protected:
+        FabricTableImpl & mProvider;
+        FabricIndex mFabric  = kUndefinedFabricIndex;
+        EndpointId mEndpoint = kInvalidEndpointId;
+        EntryIndex mNextEntryIdx;
+        EntryIndex mEntryIndex = 0;
+        uint8_t mTotalEntries  = 0;
+        uint16_t mMaxPerFabric;
+        uint16_t mMaxPerEndpoint;
+    };
+
+    uint16_t mMaxPerFabric;
+    uint16_t mMaxPerEndpoint;
+    EndpointId mEndpointId                     = kInvalidEndpointId;
+    chip::PersistentStorageDelegate * mStorage = nullptr;
+    ObjectPool<EntryIteratorImpl, kIteratorsMax> mEntryIterators;
```

Comment:
This looks unused?

Reply:
Woops sorry the latest push removed this